run.sh

```diff
@@ -0,0 +1,171 @@
+        cd $COMPONENTS_DIR/$component
+
+        # Check if versions.env file exists
+        if [ ! -f $COMPONENTS_DIR/updater/versions.env ]; then
+            # Try to get the current version from the git tags
+            current_version=$(git describe --tags --exact-match 2>/dev/null || true)
+
+        else
+            current_version=$(grep -E "^${component_upper}_VERSION=" /home/system/components/updater/versions.env | cut -d '=' -f2)
+        fi        
+
+        # echo "Current version: $current_version"
+        
+        # echo "Available versions:"
+        # git tag -l | sort -V
+
+        # Get the latest version
+        latest_version=$(git tag -l | sort -V | tail -1)
+        # echo "Latest version: $latest_version"
+
+        # Get the next version
+        next_version=$(git tag -l | sort -V | grep -A1 "$current_version" | tail -1)
+
+        # echo "Next version: $next_version"
+
+        # If there is a current version, do further checks
+        if [ "$current_version" != "" ]; then
+            # If the current version is the latest version, there are no updates
+            if [ "$current_version" = "$latest_version" ]; then
+                next_version=""
+            fi
+
+             # If there is a next version (next version is not empty or is different from current version), save it to the targets.env file
+            if [ "$next_version" != "" ]; then
+                echo "${component_upper}_VERSION=$next_version" >> /home/system/components/updater/targets.env
+            fi
+        fi
+
+        echo "$component_upper -> Current: $current_version, Next: $next_version, Latest: $latest_version"
+
+    done
+
+    # If there are no updates, exit
+    if [ ! -f $COMPONENTS_DIR/updater/targets.env ]; then
+        echo "\nNo updates available"
+    else
+        echo "\nUpdates available:"
+        cat /home/system/components/updater/targets.env
+    fi
+}
+
+# Update the system. This function is called when the --update flag is passed
+# It should do the following:
+# 1. Open the file targets.env and read the next version for each component (ws, api, apps, logger, edgeboxctl)
+# 2. Go to each component folder and checkout the next version
+# 3. Run the update script for each component
+# 4. Update the versions.env file with the new versions
+update() {
+    echo "Updating the system..."
+
+    # Check if the targets.env file exists
+    if [ ! -f targets.env ]; then
+        echo "No updates available. Run ./run.sh --check to check for updates"
+        exit 0
+    fi
+
+    # Update each component
+    for component in ws api apps logger edgeboxctl; do
+        # Get the next version from the targets.env file
+        next_version=$(grep -E "^${component_upper}_VERSION=" /home/system/components/updater/targets.env | cut -d '=' -f2)
+
+        # Go to the component folder
+        cd /home/system/components/$component
+
+        # Checkout the next version
+        git checkout $next_version
```

Comment:
What happens if the working directory is dirty? should we do anything here?

Reply:
Good point. If its dirty we probably don't want to update anything. 
Wondering if we should fully abort or just skip the component.

I guess it could be just the component skip, since if you have a dirty directory it means you're developing on it, and if you're a developer and pushed an update action, you're consciously doing it at your own risk.